### PR TITLE
first stab at myft caching in sw

### PR DESCRIPTION
**Pull request by wheresrhys**
@Financial-Times/next-myft @ironsidevsquincy (cc @adgad does the personalised caches approacj look sensible and reusable for ads?)

### CODE REVIEW

File: src/utils/personal.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const sessionToken = null;
+const personalCaches = [];
+
+const getSessionToken (cookie) {
+	return (/^.*FTSession=([^;]+).*$/.match(cookie) || [])[1];
+}
+
+self.on('fetch', ev => {
```
Comment (ironsidevsquincy) [inline src/utils/personal.js]: `toolbox.router.get('/', ..., { origin: /^https:\/\/(www|next)\.ft\.com$/ })`

?

Comment (wheresrhys) [inline src/utils/personal.js]: I don't want this to deliver a response, so easier and possibly clearer to use the low level apis.

File: src/lib/myft.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+};
+
+function clearCache (request) {
+	const networkResponse = fetch(request);
+	let resource = request.url.replace('/user', '');
+
+	caches.open('next:myft').then(cache => {
+		cache.keys().then(keys => keys.forEach(key => {
+			if (resource.indexOf(key.url) === 0) {
```
Comment (phamann) [inline src/lib/myft.js]: `if(keys.some(key => key.url === resource))`

File: src/lib/myft.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+				cache.delete(key);
+			}
+		}))
+	})
+
+	return networkResponse
+}
+
+toolbox.router.get('/__myft/api/*', cacheFirst('swMyftCaching'), cacheOptions);
```
Comment (phamann) [inline src/lib/myft.js]: `swMyftCahing` doesn't match `next:myft`?

Comment (wheresrhys) [inline src/lib/myft.js]: that's the name of the flag to turn on the cache. I made a wrapped version of sw-toolbox that only uses a cache if a flag is on

File: src/lib/myft.js (hunk 2 above)
Comment (wheresrhys) [inline src/lib/myft.js]: :+1/2:

File: src/lib/myft.js (hunk 3 above)
Comment (wheresrhys) [inline src/lib/myft.js]: the cache name only appears in cacheOptions

Comment (phamann) [inline src/lib/myft.js]: 👍

**Pull request merged**